content/en/blog/_posts/2022-07-27-sig-apps-GA-1.25/index.md

```diff
@@ -0,0 +1,80 @@
+This helps to minimize the downtime of the DaemonSet to other consumers.
+
+These features are already available in a Deployment and other workloads. This graduation helps to align this functionality across the workloads.
+
+## What problems does these features solve?
+
+### MinReadySeconds for StatefulSets
+`minReadySeconds` ensures that the StatefulSet workload is `Ready` for the given number of seconds before calling the
+pod `Available`. The notion of being `Ready` and `Available` is quite important for workloads. For example, some workloads like Prometheus with multiple instances of Alertmanager should be considered `Available` only when the Alertmanager's state transfer is complete. `minReadySeconds` also helps when using loadbalancers with cloud providers. Since the pod should be `Ready` for the given number of seconds, it provides buffer time to prevent killing pods in rotation before new pods show up.
+
+
+### MaxSurge for DaemonSets
+Kubernetes system-level components like CNI, CSI are typically run as DaemonSets. These components can have impact on the availability of the workloads if those DaemonSets go down momentarily during the upgrades. The feature allows DaemonSet pods to surge, thereby ensuring zero-downtime for the DaemonSets.
+
+Please note that the usage of `HostPort` in conjunction with `MaxSurge` in DaemonSets is not allowed as DaemonSet pods are tied to a single node and two active pods cannot share the same port on the same node.
+
+
+## How does it work?
+
+### MinReadySeconds for StatefulSets
+
+The StatefulSet controller watches for the StatefulSet pods and counts how long a particular pod has been in the `Running` state, if this value is greater than or equal to the time specified in `.spec.minReadySeconds` field of the StatefulSet, the StatefulSet controller updates the `AvailableReplicas` field in the StatefulSet's status.
+
+
+### MaxSurge for DaemonSets
+
+The DaemonSet controller creates the additional pods (above the desired number resulting from DaemonSet spec) based on the value given in `.spec.strategy.rollingUpdate.maxSurge`. The additional pods would run on the same node where the old DaemonSet pod is running till the old pod gets killed. This value cannot be `0` when `MaxUnavailable` is 0. The default value is 0.
+
+## How do I use it?
+
+### MinReadySeconds for StatefulSets
+
+Specify a value for `minReadySeconds` for any StatefulSet and check if pods are available or not by inspecting
+`AvailableReplicas` field using:
+
+`kubectl get statefulset/<name_of_the_statefulset> -o yaml`
+
+Please note that the default value of `minReadySeconds` is 0.
+
+### MaxSurge for DaemonSets
+
+Specify a value for `.spec.updateStrategy.rollingUpdate.maxSurge` and set `.spec.updateStrategy.rollingUpdate.maxUnavailable` to `0`. 
+
+Then observe a faster rollout and higher number of pods running at the same time in the next rollout.
+
+```
+oc rollout restart daemonset <name_of_the_daemonset>
```

Comment:
Is this an openshift-specific feature? If not, then could you show this with a kubectl command instead of an oc command?

Reply:
It is not openshift specific. I forgot to switch it back to kubectl